jwst/tweakreg/tweakreg_step.py

```diff
@@ -315,53 +323,53 @@ def process(self, input):
-                            max_inv_pix_error=self.sip_max_inv_pix_error,
-                            inv_degree=self.sip_inv_degree,
-                            npoints=self.sip_npoints,
-                            crpix=None
-                        )
-                    except (ValueError, RuntimeError) as e:
-                        msg = f"Failed to update 'meta.wcsinfo' with FITS SIP \
-                            approximation. Reported error is: \n {e.args[0]}"
-                        self.log.warning(msg)
-            record_step_status(image_model, "tweakreg", success=True)
-
-        return image_model
+                        ) -> ModelLibrary:
+        with images:
+            for (image_model, corrector) in zip(images, correctors):
+
+                # retrieve fit status and update wcs if fit is successful:
+                if ("fit_info" in corrector.meta and
+                        "SUCCESS" in corrector.meta["fit_info"]["status"]):
+
+                    # Update/create the WCS .name attribute with information
+                    # on this astrometric fit as the only record that it was
+                    # successful:
+                    if align_to_abs_refcat:
+                        # NOTE: This .name attrib agreed upon by the JWST Cal
+                        #       Working Group.
+                        #       Current value is merely a place-holder based
+                        #       on HST conventions. This value should also be
+                        #       translated to the FITS WCSNAME keyword
+                        #       IF that is what gets recorded in the archive
+                        #       for end-user searches.
+                        corrector.wcs.name = f"FIT-LVL3-{self.abs_refcat}"
+
+                    image_model.meta.wcs = corrector.wcs
+                    update_s_region_imaging(image_model)
+
+                    # Also update FITS representation in input exposures for
+                    # subsequent reprocessing by the end-user.
+                    if self.sip_approx:
+                        try:
+                            update_fits_wcsinfo(
+                                image_model,
+                                max_pix_error=self.sip_max_pix_error,
+                                degree=self.sip_degree,
+                                max_inv_pix_error=self.sip_max_inv_pix_error,
+                                inv_degree=self.sip_inv_degree,
+                                npoints=self.sip_npoints,
+                                crpix=None
+                            )
+                        except (ValueError, RuntimeError) as e:
+                            msg = f"Failed to update 'meta.wcsinfo' with FITS SIP \
+                                approximation. Reported error is: \n {e.args[0]}"
+                            self.log.warning(msg)
+                images.shelve(image_model)
+        record_step_status(images, "tweakreg", success=True)
```

Comment:
```suggestion
        record_step_status(images, "tweakreg", success=True)
```
Having this outside the loop will result in a second iteration through the models for this step. Any objection to moving it back inside the loop?